src/sparseml/openpifpaf/train.py

```diff
@@ -151,11 +152,35 @@ def main():
 
     datamodule = openpifpaf.datasets.factory(args.dataset)
 
+    """
+    Cases for checkpointing:
+    1. No checkpoint - no changes to base flow
+    2. Checkpoint from local file - no changes to base flow
+    3. Checkpoint from zoo stub:
+        a. set args.checkpoint to None to get a base net
```

Comment:
how is the base net specified? does the user need to include that as CLI args? If so, let's look into including the specification in the state dict to remove that potential point of user failure

Reply:
--basenet from cli. it is also saved into the checkpoint under checkpoint["meta"]["args"]["basenet"]. For loading from a checkpoint we read basenet from the checkpoint. For normal training users have to specify on CLI